### DISCUSSION TITLE
Crazyflie AI Deck Connect to Wifi

### DISCUSSION THREAD

**Discussion by laurenbramblett** [bitcraze/discussions#892] [Q&A]
Hello, I would like to connect my crazyflie to our lab's wifi and I have set it up via 
```
make menuconfig
```
in crazyflie-firmware. I then flashed the cf2.bin to the crazyflie using 
```
make cload
```
I then tried flashing the gap8 bootloader again over wifi and via JTAG, but am still unable to find my device on the list of connected devices on the router and this is the output from the cfclient:
unning!
SYS: Build 14:e176751cf1ec (2023.07 +14) CLEAN
SYS: I am 0x203430345242500A003D0041 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcAI for deck 0
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized, CPX_VERSION: 0
AIDECK: AI-deck will connect to WiFi
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.02
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 14600 bytes
CPX: ESP32: I (435) SYS: Initialized
CPX: ESP32: I (435) DISCOVERY: Hostname is aideck-E79784
CPX: ESP32: I (435) DISCOVERY: MAC (STA) is c8:c9:a3:e7:97:84
CPX: GAP8: -- WiFi image streamer example --
CPX: ESP32: I (2725) WIFI: wifi_init_sta finished.
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly
CPX: GAP8: Setting up WiFi AP
CPX: GAP8: Starting camera task...
CPX: ESP32: I (3875) WIFI: wifi_init_softap finished

How do I get the crazyflie to connect and stream over wifi? I have tried to view the camera stream with the following command:
```
python opencv-viewer.py -n aideck-E79784.local
```
But I got the error: " Name or service not known"

Any help is appreciated!

**Comment by evoggy** [comment]
Hi,

It seems as if it's trying to set up both an access point and connecting to a WiFi and this will not work. The AI deck is shipped with an example where it automatically set's up an access point (to quickly get started). So in order to switch to connecting to your own WiFi you also have to re-flash the ESP without this. The default config on master for the ESP firmware will not have this features, so if you clone this, build it and flash it then you should be able to connect to it.

**Comment by laurenbramblett** [comment]
Hi!

Do you mean the code found here?
https://github.com/bitcraze/aideck-esp-firmware

Thanks!

**Comment by laurenbramblett** [comment]
As a follow up, how would I switch back to using the crazyflie as an access point? Would I just follow the instructions here:
https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-the-latest-esp-firmware

**Comment by laurenbramblett** [comment]
After flashing the esp-firmware over a JTAG, I am still having the same problem. Is there a step-by-step solution I could follow? The network "Wifi-streaming-example" is still discoverable

**Comment by evoggy** [comment]
After digging into this a bit more I think the issue is in the GAP8. The default GAP8 example is the one setting up the WiFi access point, if you can rebuild the wifi image streamer for the GAP8 and flash it then it should stop setting up the WiFi.
